xbmc/threads/Thread.h

```diff
@@ -36,7 +36,16 @@ class CThread
   CThread(IRunnable* pRunnable, const char* ThreadName);
   virtual ~CThread();
   void Create(bool bAutoDelete = false);
-  void Sleep(unsigned int milliseconds);
+
+  template<typename Rep, typename Period>
+  void Sleep(std::chrono::duration<Rep, Period> duration)
+  {
+    if (duration > std::chrono::duration<Rep, Period>(10) && IsCurrentThread())
```

Comment:
This is not the same anymore. If the `std::chrono::duration` is e.g. `std::chrono::seconds` you will now check agains `10s` instead of against `10ms`. So you need `duration > std::chrono::milliseconds(10)`.

Reply:
Thanks for catching this. I actually had it like that originally then changed it in the last round of updates.